python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+            raise RuntimeError("Butler query found %s for runs, but the run specified (%s) "
+                               "was not among them."%(runs, run))
+        elif run is None:  # we know it's OK now
+            run = butler.queryMetadata('raw', ['run'])[0]
+        del runs  # we have run defined now, so remove this to avoid potential confusion later
+
+        if not os.path.exists(self.config.eotestOutputPath):
+            os.mkdir(self.config.eotestOutputPath)
+
+        ccds = butler.queryMetadata('raw', ['ccd'])
+        imTypes = butler.queryMetadata('raw', ['imageType'])
+        testTypes = butler.queryMetadata('raw', ['testType'])
+
+        ################################
+        ################################
+        # Run the Fe55 task
+        if self.config.doFe55:
+            fe55TaskDataId = {'run': run, 'testType': 'FE55', 'imageType': 'FE55'}
+            self.log.info("Starting Fe55 pixel task")
+            for ccd in ccds:
+                if 'FE55' not in testTypes:  # TODO: Remove not to test functionality
+                    msg = "No Fe55 tests found. Available data: %s"%testTypes
+                    if self.config.requireAllEOTests:
+                        raise RuntimeError(msg)
+                    else:
+                        self.log.warn(msg + "\nSkipping Fe55 task")
+                        break
+                fe55Filenames = [butler.get('raw_filename', dataId={'visit': visit,
+                                                                    'ccd': ccd})[0][:-3]
+                                 for visit in butler.queryMetadata('raw', ['visit'], dataId=fe55TaskDataId)]
+                self.log.trace("Fe55Task: Processing %s with %s files"%(ccd, len(fe55Filenames)))
+                maskFiles = self._getMaskFiles(self.config.eotestOutputPath, ccd)
+                gains = self.fe55.run(sensor_id=ccd, infiles=fe55Filenames, mask_files=maskFiles)
+                gainsPropSet = dafBase.PropertySet()
+                for amp, gain in gains.items():  # there is no propSet.fromDict() method so make like this
+                    gainsPropSet.addDouble(str(amp), gain)
+                butler.put(gainsPropSet, 'eotest_gain', dataId={'ccd': ccd, 'run': run})
+            del fe55TaskDataId
+
+        # TODO: validate the results above, and/or change code to (be able to) always run
+        # over all files instead of stopping at the "required accuracy"
+        # This will require making changes to the eotest code.
+
+        ################################
+        ################################
+        # Run the Noise task
+        if self.config.doReadNoise:
+            # self.readNoise.config.output_dir = '/home/mfl/thisisbad/'
+            # note that LCA-10103 defines the Fe55 bias frames as the ones to use here
+            self.log.info("Starting readNoise task")
+            noiseTaskDataId = {'run': run, 'testType': 'FE55', 'imageType': 'BIAS'}
```

Comment:
Why do images have `testType=FE55` but `imageType=BIAS`? There's a fundamental difference between Fe55 images and bias images. Wouldn't it make more sense to just have `imageType=FE55`?

Reply:
There's two things to note here. The first is that the intent of the people writing the files was to note that these are bias frames taken during an Fe55 test run, so the _testType_ is Fe55, but the _imageType_ is BIAS.

The second point to note is that this code exists to interface with a pre-defined dataset structure which we have no control over, and so this is just dealing with the way things are.